addons/base_import/models/base_import.py

```diff
@@ -113,20 +142,56 @@ def _can_import_remote_urls(self):
+        A.4. Try to find for each header a field to map with (see "_get_mapping_suggestions")
+            - First check the previously saved mappings between the header name and one of the model's fields
+            - If no mapping found, try an exact match comparison using fields technical names, labels and user language
+              translated labels
+            - If nothing found, try a fuzzy match using word distance between header name and fields tachnical
+              names, labels and user language translated labels. Keep only the closest match.
+        A.5. Prepare examples for each columns using the first non null value from each column.
+        A.6. Send the info back to the UI where the user can modify the suggested mapping.
+    B. Execute the import: There are two import mode with uses the same process. (see "execute_import")
+        B.1. Test import: Try to import but rollback the transaction. This allows the check errors during the import
+                          process and allow the user to choose import options for the different encountered errors.
+        B.2. Real import: Try to import the file using the configured mapping and the eventual "error mapping options".
+                          If import encounters blocking errors, the transaction is rollbacked and the user is allowed
+                          to choose import options for the different errors.
+            - Get file data and fields to import into (see "_convert_import_data")
+            - Parse date, float and binary data (see "_parse_import_data")
+            - Handle multiple mapping -> concatenate char/text/many2many columns mapped on the same field
+              (see "_handle_multi_mapping")
+            - Handle fallback values for boolean and selection fields, in case input data does not match any allowed values.
+              (see "_handle_fallback_values")
+            - Load data (see ir.model "load" method)
+            - Rollback transaction if test mode or if encountered error.
+            - Save mapping if any import is successful to ease later mapping suggestions
+            - Return import result to the UI (success or errors if any)
+    """
 
     _name = 'base_import.import'
     _description = 'Base Import'
 
     # allow imports to survive for 12h in case user is slow
     _transient_max_hours = 12.0
+    # we consider that if the difference is more than 0.2, then the two compared strings are "too different" to propose
+    # any match between them. (see '_get_mapping_suggestion' for more details)
+    FUZZY_MATCH_DISTANCE = 0.2
 
     res_model = fields.Char('Model')
     file = fields.Binary('File', help="File to check and/or import, raw binary (not base64)", attachment=False)
     file_name = fields.Char('File Name')
     file_type = fields.Char('File Type')
 
     @api.model
-    def get_fields(self, model, depth=FIELDS_RECURSION_LIMIT):
+    def get_fields_tree(self, model, depth=FIELDS_RECURSION_LIMIT):
```

Comment:
Why did you rename this method?
BTW technically speaking, the returned value is not a *tree* but a *forest* (collection of trees).